okcupyd/profile.py

```diff
@@ -213,14 +215,25 @@ def essays(self):
         """
         return essay.Essays(self)
 
+    _age_xpb = xpb.span.with_class('userinfo2015-basics-asl-age')
+    _user_age_xpb = xpb.span(id='ajax_age')
+
     @util.cached_property
     def age(self):
         """
         :returns: The age of the user associated with this profile.
         """
-        return int(xpb.span(id='ajax_age').get_text_(self.profile_tree).strip())
 
-    _percentages_and_ratings_xpb = xpb.div(id='percentages_and_ratings')
+        try:
+            #Attempt to get a non logged-in user's profile age
+            profile_age = int(self._age_xpb.get_text_(self.profile_tree))
+        except:
```

Comment:
Bare excepts are bad because they can mask ohter types of exceptions. Lets pick a specific exception here.